### ISSUE TITLE
API Questions for iD

### ISSUE THREAD

**Issue by SilentSpike** [osm-fr/osmose-frontend#193]
So I'm currently working to add an Osmose data layer into iD (https://github.com/openstreetmap/iD/pull/7095) and have ran into a few questions:
1. I notice the 3.0 API is in beta, currently the code I've written is using this endpoint, but should I be using the 2.0 endpoint until it is out of beta or is the output likely to be pretty stable?
2. Sort of tied into the previous question, I notice the 3.0 API has stopped returning "elems" when using parameter `full=true`. Is this an intentional change? I see that it was changed here: https://github.com/osm-fr/osmose-frontend/commit/73274d62a562fa3dc46fd66710405e265a61247f#diff-1c7c3f69f18525bc18d14e58eea66df6L108
3. Are there any considerations I should have when using the API? I imagine a minority of iD users are actively using the external QA data layers, but even this could still lead to a noticeable increase in usage of the API so I'd like to be considerate.
    * The editor will be caching errors so as to not repeat requests unnecessarily and I'll be using the `item=` parameter to only requests the types of errors we want to show users of iD. It also only requests errors when the user is zoomed in to level 14 or above.
    * Unfortunately I need to use the `full=true` parameter because iD needs both the `item` and `class` of errors to determine what they are (used for rendering and title/description), as well as the associated `elems` (used for dynamic description content and element highlighting).

Another minor thing I noticed is that you're giving the class with an extra s:
https://github.com/osm-fr/osmose-frontend/blob/e71817de24fe3db72f3f954ced6cd0491deb7e59/errors.py#L107

**Comment by frodrigo** [comment]
Glad to know about this project.

The 0.3beta changed this week with an internal refactoring. The plan is now to complete the 0.3beta API and release it as 0.3. Then deprecate the 0.2 API. The 0.2 API is already at somewhat a wrapper around the 0.3beta.

Rather than using /api/0.3beta/issues you could use the tiles /map/issues/<z:int>/<x:int>/<y:int>.<format:ext>. The tile have to be moved into the API. The tiles accept parameters and quickly to provide.

The `elems` properties was an internal way to deduplicates issues. It is now removed. Deduplications is now done by unique stable uuid. The objects part of issue can be found under issue details http://osmose.openstreetmap.fr/fr/api/0.3beta/issue/638a7418-851c-7177-0731-ccf20ee89125 Item and class are on tiles.

If you can, I suggest you to not filter on item/class number but rather on level and/or on osmose tags.

I also suggest you to avoid translation on your part. The API can be call with a language and return translation. I am also working on the ability the serve (translated) doc/help on item/class in the API 0.3beta.

Please ask for any required change in the API 0.3beta. The web part of osmose should be separated from API and it is the right time to do it.

QA tools have to be use with precaution, particularly with new users. There may be misuse or abuse.

Thank for reporting the typo on `classs`.

**Comment by SilentSpike** [comment]
Thank you for the quick response, very helpful information 👍 

I've tweaked my code now to use the tiles - it actually makes my life slightly easier as well as cutting down the unnecessary information sent.

Good to know that `elems` has been removed and I was using it in an unintended way 😄 I can make it so that when the UI is loaded a request is made to get the specific issue details. On that point, iD doesn't need all the tag information, not sure if it's worth your time (in terms of reducing overhead or data usage) providing a way to specify that.

For filtering, I see issues in all 3 levels and most osmose tags that iD can use. Would it be better for you if I make requests without filtering and filter out issues by `item` on iD's end?

As for using the supplied translations, I'll have to discuss with the actual iD developers as I think a lot of them may not be worded in the way iD typically presents users with information (and doesn't allow us to directly insert element links into the description itself).
